lib/index.js

```diff
@@ -1,40 +1,65 @@
 var winston = require('winston');
+var utils = require('./utils');
+var nodeUtil = require('util');
 
-var logger = new winston.Logger();
+function Logger() {}
 
-var utils = require('./utils');
+function createInstance() {
+    if (global.sharedLogger) {
+        return global.sharedLogger
+    } else {
+        nodeUtil.inherits(Logger, winston.Logger);
 
-logger.setup = function setup(config, setMasterConfig) {
-    if (setMasterConfig || !this.hasMasterConfig) {
-        this.setConfig(config, setMasterConfig);
-    }
-};
+        Logger.prototype.setup = function setup(config, setMasterConfig) {
+            if (!this.hasMasterConfig) {
+                this.configure({
+                    transports : [new winston.transports.Console({
+                        label : 'not-set',
+                        debugStdout: true,
+                        timestamp: utils.timestamp,
+                        formatter: utils.formatter
+                    })]
+                });
 
-logger.setConfig = function(config, setMasterConfig) {
+                this.level = config && config.loglevel ? config.loglevel : 'debug';
+            }
 
-    if (setMasterConfig) {
-        this.hasMasterConfig = true;
-    }
+            if (config && (setMasterConfig || !this.hasMasterConfig)) {
+                this.setConfig(config, setMasterConfig);
+            }
+
+            return this;
+        };
 
-    this.teardown();
+        Logger.prototype.setConfig = function(config, setMasterConfig) {
+            if (setMasterConfig) {
+                this.hasMasterConfig = true;
+            }
 
-    config = config || {};
-    this.level = config.loglevel || 'debug';
+            this.teardown();
 
-    this.add(winston.transports.Console, (config.transports && config.transports.console) || {
-        debugStdout: true,
-        label: config.appName || 'not-set',
-        timestamp: utils.timestamp,
-        formatter: utils.formatter
-    });
-};
+            if (config.logLevel) {
+                this.level = config.logLevel;
+            }
+
+            this.add(winston.transports.Console, (config.transports && config.transports.console) || {
+                    debugStdout: true,
+                    label: config.appName || 'not-set',
+                    timestamp: utils.timestamp,
+                    formatter: utils.formatter
+                });
+        };
+
+        Logger.prototype.teardown = function teardown() {
+            Object.keys(this.transports).forEach(function(transport) {
+                this.remove(transport);
+            }.bind(this));
+        };
+
+        return global.sharedLogger = new Logger().setup();
+    }
+}
 
-logger.teardown = function teardown() {
-    Object.keys(this.transports).forEach(function(transport) {
-        this.remove(transport);
-    }.bind(this));
-};
+module.exports = createInstance();
```

Comment:
I took a look at http://justjs.com/posts/singletons-in-node-js-modules-cannot-be-trusted-or-why-you-can-t-just-do-var-foo-require-baz-init to compare the approach and it's practically the same :)

+1